### PR TITLE
Use heredoc to make command help source nicer

### PR DESCRIPTION
**Pull request by simonbaird**
Also add some line breaks in some of the longer flag descriptions and some extra blank lines for consistency and readability.

### CODE REVIEW

**Comment by simonbaird** [comment]
You could argue against using the heredoc trimmer like this for the shorter strings, but I decided to do it anyway for the sake of consistency and tidiness.

**Comment by zregvart** [comment]
I like the intent, I'm not a fan of the execution. The problem I see is that we need to remember to use heredoc for commands. What I think we should do instead is to modify the help/usage templates so they apply universally. Here is what I did instead by copying the existing template and adding the `| heredoc` here and there:

```diff
diff --git a/cmd/root.go b/cmd/root.go
index 848bbd3..fdf6079 100644
--- a/cmd/root.go
+++ b/cmd/root.go
@@ -21,6 +21,7 @@ import (
        "os"
        "time"
 
+       "github.com/MakeNowJust/heredoc"
        hd "github.com/MakeNowJust/heredoc"
        "github.com/spf13/cobra"
 
@@ -66,6 +67,41 @@ func init() {
                same as verbose but also show function names and line numbers
        `))
 
+       cobra.AddTemplateFunc("heredoc", heredoc.Doc)
+       RootCmd.SetHelpTemplate(`{{with (or .Long .Short)}}{{. | heredoc}}
+
+{{end}}{{if or .Runnable .HasSubCommands}}{{.UsageString}}{{end}}`)
+       RootCmd.SetUsageTemplate(`Usage:{{if .Runnable}}
+  {{.UseLine | heredoc}}{{end}}{{if .HasAvailableSubCommands}}
+  {{.CommandPath}} [command]{{end}}{{if gt (len .Aliases) 0}}
+
+Aliases:
+{{.NameAndAliases}}{{end}}{{if .HasExample}}
+
+Examples:
+{{.Example | heredoc}}{{end}}{{if .HasAvailableSubCommands}}{{$cmds := .Commands}}{{if eq (len .Groups) 0}}
+
+Available Commands:{{range $cmds}}{{if (or .IsAvailableCommand (eq .Name "help"))}}
+{{rpad .Name .NamePadding }} {{.Short}}{{end}}{{end}}{{else}}{{range $group := .Groups}}
+
+{{.Title}}{{range $cmds}}{{if (and (eq .GroupID $group.ID) (or .IsAvailableCommand (eq .Name "help")))}}
+{{rpad .Name .NamePadding }} {{.Short}}{{end}}{{end}}{{end}}{{if not .AllChildCommandsHaveGroup}}
+
+Additional Commands:{{range $cmds}}{{if (and (eq .GroupID "") (or .IsAvailableCommand (eq .Name "help")))}}
+{{rpad .Name .NamePadding }} {{.Short}}{{end}}{{end}}{{end}}{{end}}{{end}}{{if .HasAvailableLocalFlags}}
+
+Flags:
+{{.LocalFlags.FlagUsages | trimTrailingWhitespaces}}{{end}}{{if .HasAvailableInheritedFlags}}
+
+Global Flags:
+{{.InheritedFlags.FlagUsages | trimTrailingWhitespaces}}{{end}}{{if .HasHelpSubCommands}}
+
+Additional help topics:{{range .Commands}}{{if .IsAdditionalHelpTopicCommand}}
+{{rpad .CommandPath .CommandPathPadding}} {{.Short}}{{end}}{{end}}{{end}}{{if .HasAvailableSubCommands}}
+
+Use "{{.CommandPath}} [command] --help" for more information about a command.{{end}}
+`)
+
        kubernetes.AddKubeconfigFlag(RootCmd)
 }
```

We'd need to do the same for the YAML generation in `internal/documentation/documentation.go` though.

**Comment by simonbaird** [comment]
Yeah okay. I'll give it a try.

File: cmd/create_pr.go [hunk 1]
```diff
@@ -35,32 +36,64 @@ func createPR() *cobra.Command {
...
 		},
 	}
-	cmd.Flags().StringVar(&data.componentRepoURL, "repo", data.componentRepoURL, "repo URL")
-	cmd.Flags().StringVar(&data.destinationBranch, "destination-branch", data.destinationBranch, "target branch for PR")
-	cmd.Flags().StringVar(&data.prBranchName, "branch-name", data.prBranchName, "branch to create with patch changes")
-	cmd.Flags().StringVar(&data.prTitle, "title", data.prTitle, "title of the PR")
-	cmd.Flags().StringVar(&data.prBody, "body", data.prBody, "body of the PR")
-	cmd.Flags().StringVar(&data.patchFilePath, "patch", data.patchFilePath, "path to the patch file")
+
+	cmd.Flags().StringVar(&data.componentRepoURL, "repo", data.componentRepoURL, hd.Doc(`
```
Comment (lcarva) [inline cmd/create_pr.go:73]: `hd.Doc` for single line strings feels heavy handed.

**Review by lcarva — APPROVED** [review]

**Review by zregvart — APPROVED** [review]
Merge this if you need it, we're discussing aesthetics :)

**Comment by simonbaird** [comment]
I made it less hard-core, i.e. don't use the heredoc for single line strings, which includes most of the flag docs.

**Comment by simonbaird** [comment]
We discussed this at standup, and agreed it's okay to merge. I'll self-review carefully and merge soon probably.

**Pull request merged**